File: app/src/main/java/im/dacer/kata/util/engine/SearchEngine.java
package im.dacer.kata.util.engine;

import android.content.Context;

import im.dacer.kata.util.action.BingSearchAction;
import im.dacer.kata.util.action.DuckDuckGoSearchAction;
import im.dacer.kata.util.action.GoogleSearchAction;
import im.dacer.kata.util.action.JishoSearchAction;
import im.dacer.kata.util.action.SearchAction;
import im.dacer.kata.data.local.MultiprocessPref;

public class SearchEngine {

    public static final String GOOGLE = "Google";
    public static final String BING = "Bing";
    public static final String DUCKDUCKGO = "DuckDuckGo";
    public static final String JISHO = "Jisho";

    public static SearchAction getDefaultSearchAction(Context context) {
        MultiprocessPref multiprocessPref = new MultiprocessPref(context);
        return getSearchAction(multiprocessPref.getSearchEngine());
    }

    public static SearchAction getSearchAction(int index) {
        switch (getSupportSearchEngineList()[index]) {
            case GOOGLE:
                return GoogleSearchAction.create();
            case BING:
                return BingSearchAction.create();
            case DUCKDUCKGO:
                return DuckDuckGoSearchAction.create();
            case JISHO:
                return JishoSearchAction.create();
        }
        return null;
    }

    public static SearchAction getSearchAction(String engine) {
        switch (engine) {
            case GOOGLE:
                return GoogleSearchAction.create();
            case BING:
                return BingSearchAction.create();
            case DUCKDUCKGO:
                return DuckDuckGoSearchAction.create();
            case JISHO:
                return JishoSearchAction.create();
        }
        return null;
    }

    public static String[] getSupportSearchEngineList() {
        return new String[]{GOOGLE, BING, DUCKDUCKGO, JISHO};
    }
}

File: app/src/main/java/im/dacer/kata/util/engine/SearchEngine.kt
package im.dacer.kata.util.engine

import android.content.Context
import im.dacer.kata.data.local.MultiprocessPref
import im.dacer.kata.util.action.*

object SearchEngine {

    const val GOOGLE = "Google"
    const val BING = "Bing"
    const val DUCKDUCKGO = "DuckDuckGo"
    const val JISHO = "Jisho"
    const val MOJI = "Moji"

    val supportSearchEngineList: Array<String>
        get() = arrayOf(GOOGLE, BING, DUCKDUCKGO, JISHO, MOJI)

    fun getDefaultSearchAction(context: Context): Action? {
        val multiprocessPref = MultiprocessPref(context)
        return getSearchAction(multiprocessPref.searchEngine)
    }

    fun getSearchAction(engine: String): Action? {
        when (engine) {
            GOOGLE -> return GoogleSearchAction.create()
            BING -> return BingSearchAction.create()
            DUCKDUCKGO -> return DuckDuckGoSearchAction.create()
            JISHO -> return JishoSearchAction.create()
            MOJI -> return MojiSearchAction.create()
        }
        return null
    }
}
